### PR TITLE
switch from enum to string based document types

### PR DESCRIPTION
**Pull request by garyhgohoos**
Please give this change a thorough code review

### CODE REVIEW

File: widgets/comment.cpp [hunk 1]
```diff
@@ -388,120 +218,81 @@ void comment::set(const ParameterList &pParams)
   param = pParams.value("sourceType", &valid);
   if (valid)
   {
-    _source = (enum Comments::CommentSources)param.toInt();
+    _sourcetype = param.toString();
     if(!(_mode == cEdit || _mode == cView))
     {
-      switch (_source)
```
Comment (gpazo) [inline widgets/comment.cpp]: Not sure I agree with removing this but if it's easier to read, carry on

File: widgets/comments.cpp [hunk 2]
```diff
@@ -159,9 +203,21 @@ Comments::Comments(QWidget *pParent, const char *name) :
...
 
-void Comments::setType(enum CommentSources pSource)
+int Comments::type() const
 {
-  _source = pSource;
+  CommentMap *elem = _strMap.value(_sourcetype);
+  return elem ? elem->doctypeId : Uninitialized;
+}
+
+void Comments::setType(int sourceType)
```
Comment (gpazo) [inline widgets/comments.cpp]: 2 functions, same name but different input types?

Comment (davidbeauchamp) [inline widgets/comments.cpp]: Also known as an overload

File: widgets/comment.cpp (hunk 1 above)
Comment (gilmoskowitz) [inline widgets/comment.cpp]: If we're going to use string comparisons, the cascading else/if is required in C++. It happens to be much slower than an integer-based switch.

A more extensible way, faster than the else/if but slower than the switch, would be to do a lookup from the `QMap<QString, struct CommentMap*> _strMap` from comments.cpp below.

File: widgets/comments.cpp [hunk 3]
```diff
@@ -38,55 +38,99 @@ void Comments::showEvent(QShowEvent *event)
...
+//    addToMap(TimeExpense,       "TE",    tr("Time Expense")                                     );
+    addToMap(TodoItem,          "TD",    tr("To-Do"),           "todoitem_id","todoItem",     "MaintainPersonalToDoItems MaintainAllTodoItems" );
+    addToMap(TransferOrder,     "TO",    tr("Transfer Order"),  "tohead_id",  "transferOrder"   );
+    addToMap(TransferOrderItem, "TI",    tr("Transfer Order Item")                              );
+    addToMap(Vendor,            "V",     tr("Vendor"),          "vend_id",    "vendor"          );
+//    addToMap(Voucher,           "VCH",   tr("Voucher"),         "vohead_id",  "voucher"         );
+    addToMap(Warehouse,         "WH",    tr("Site")                                             );
+    addToMap(WorkOrder,         "W",     tr("Work Order"),      "wo_id",      "workOrder"       );
+    addToMap(Task,              "TA",    tr("Project Task"),    "prjtask_id", "projectTask"     );
+  }
```
Comment (gilmoskowitz) [inline widgets/comments.cpp]: Looks like `TimeAttendance , "TATC"` is missing from the map.

**Pull request merged**